File: Python/283__Move_Zeroes.py
class Solution(object):
    def moveZeroes(self, nums):
        """
        :type nums: List[int]
        :rtype: void Do not return anything, modify nums in-place instead.
        """
        index = 0
        for i in range(len(nums)):
            if nums[i] != 0:
                if index < i:
                    nums[index] = nums[i]
                    index += 1
                    nums[i] = 0
                else:
                    index += 1


                

File: Java/283__Move_Zeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
        int save_index = 0;
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] != 0) {
                if (save_index != i) {
                    nums[save_index] = nums[i];
                }
                save_index++;
            }
        }
        for (int i = save_index; i < nums.length; i++) {
            nums[i] = 0;
        }
    }
}
